From dc4e4cc806d40a47f03023ab1aed2db946f34cb6 Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Tue, 17 Mar 2020 19:11:23 +0800
Subject: [PATCH] ARM: alignment: Call alignment_init() later when
 CONFIG_ROCKCHIP_THUNDER_BOOT=y

alignment_init() create proc file "cpu/alignment" at the fs_initcall
level. While "cpu" dir is created on proc_cpu_init() at the same
initcall level too.

Fix this by move alignment_init() to the fs_initcall_sync level.

Change-Id: I8a8831f103b6729f57a8a70ff1bf5672fdf98810
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 arch/arm/mm/alignment.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/mm/alignment.c b/arch/arm/mm/alignment.c
index 84a6bbaf8cb2..382d36169b47 100644
--- a/arch/arm/mm/alignment.c
+++ b/arch/arm/mm/alignment.c
@@ -1055,4 +1055,8 @@ static int __init alignment_init(void)
 	return 0;
 }
 
+#ifdef CONFIG_ROCKCHIP_THUNDER_BOOT
+fs_initcall_sync(alignment_init);
+#else
 fs_initcall(alignment_init);
+#endif
-- 
2.35.3

